presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/HistoricalStatisticsEquivalentPlanMarkingOptimizer.java

```diff
@@ -79,6 +86,12 @@ public PlanNode optimize(PlanNode plan, Session session, TypeProvider types, Var
             return plan;
         }
 
+        // Only enable history based optimization when plan has a join/aggregation.
+        if (restrictHistoryBasedOptimizationToComplexQuery(session) &&
+                !PlanNodeSearcher.searchFrom(plan).where(node -> PRECOMPUTE_PLAN_NODES.stream().anyMatch(clazz -> clazz.isInstance(node))).matches()) {
```

Comment:
why do we stream PRECOMPUTE_PLAN_NODES and not just use List.contains?

Reply:
This is because the `node` is an instance of PlanNode, while `PRECOMPUTE_PLAN_NODES` is a list of PlanNode type, not the instance itself.